horqrux/primitive.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import annotations
+
+from .abstract import Primitive
+from .utils import ControlQubits, TargetQubits
+
+# Single qubit gates
+
+
+def I(target: TargetQubits, control: ControlQubits = (None,)) -> Primitive:
+    """The Identity operation on a single qubit.
+
+    Args:
+        target (TargetIdx): Tuple of Tuples describing the qubits to apply to.
+        control (ControlIdx, optional): Tuple of Tuples or Nones describing
+        the control qubits of length(target). Defaults to (None,).
+
+    Returns:
+        Gate: Gate object.
+    """
+    return Primitive("I", target, control)
+
+
+def NOT(target: TargetQubits, control: ControlQubits = (None,)) -> Primitive:
+    """NOT gate. Note that since we lazily evaluate the circuit, this function
+    returns the gate representation of Gate type and does *not* apply the gate.
+    By providing a control idx it turns into a controlled gate, use None for no control qubits.
+
+    Example usage: `NOT(((1, ), ), (None, ))` applies the NOT to qubit 1.
+
+    Example usage controlled: `NOT(((1, )), ((0, )))` applies the NOT to qubit 1 with controlled bit 0.
+
+    Args:
+        target (TargetIdx): Tuple of Tuples describing the qubits to apply to.
+        control (ControlIdx, optional): Tuple of Tuples or Nones describing
+        the control qubits of length(target). Defaults to (None,).
+
+    Returns:
+        Gate: Gate object.
+    """
+    return Primitive("NOT", target, control)
+
+
+def X(target: TargetQubits, control: ControlQubits = (None,)) -> Primitive:
+    """X gate. Note that since we lazily evaluate the circuit, this function
+    returns the gate representation of Gate type and does *not* apply the gate.
+    By providing a control idx it turns into a controlled gate, use None for no control qubits.
+
+    Example usage: X(((1, ), ), (None, )) applies the NOT to qubit 1.
```

Comment:
```suggestion
    Example usage: X(((1, ), ), (None, )) applies the X to qubit 1.
```

Reply:
Please note that `CX` is identical to `CNOT`.

Reply:
fixed